### PR TITLE
DatabaseService: fix ResourceMatcher field

### PR DESCRIPTION
**Pull request by marcoandredinis**
In a previous PR (#19363) we created a new resource type: DatabaseService

Its only spec property for now is the ResourceMatcher field
This field should mirror what we offer in the configuration within `db_service.resurces` from `teleport.yaml`.

Even though, the current implementation of
`types.DatabaseService.ResourceMatchers` is convirtable to/from `services.ResourceMatchers` - because its only field (from the latter) is a list of labels - we would incur in breaking changes later on if the `services.ResourceMatchers` got new fields.

This new resource is only in master (not yet released) and the backport to v11 must include this change to prevent a breaking change from happening in the future.

### CODE REVIEW

**Review by smallinsky — APPROVED** [review]

**Comment by codingllama** [comment]
Since we are here, should we sneak in a fix for Marek's comment: https://github.com/gravitational/teleport/pull/19363#discussion_r1052239190 ?

File: api/proto/teleport/legacy/types/types.proto [hunk 1]
```diff
@@ -4429,8 +4429,13 @@ message DatabaseServiceV1 {
...
 message DatabaseServiceSpecV1 {
   // ResourceMatchers is the configured match for Database resources.
-  repeated wrappers.LabelValues ResourceMatchers = 1 [
-    (gogoproto.jsontag) = "resources",
+  repeated ResourceMatcher ResourceMatchers = 1 [(gogoproto.jsontag) = "resources"];
+}
+
+// ResourceMatcher is a set of properties that is used to match on resources.
+message ResourceMatcher {
+  wrappers.LabelValues labels = 1 [
```
Comment (greedy52) [inline api/proto/teleport/legacy/types/types.proto:4437]:
```suggestion
  wrappers.LabelValues Labels = 1 [
```
?

Also, do you think if `(gogoproto.nullable) = false, ` is needed for either of these?

**Review by greedy52 — APPROVED** [review]

File: api/proto/teleport/legacy/types/types.proto (hunk 1 above)
Comment (codingllama) [reply api/proto/teleport/legacy/types/types.proto:4437]: My 2c: don't use gogo features unless you absolutely need them, it's deprecated and we would like to move out of it sometime.

File: api/proto/teleport/legacy/types/types.proto [hunk 2]
```diff
@@ -4429,8 +4429,13 @@ message DatabaseServiceV1 {
 // DatabaseServiceSpecV1 is the DatabaseService Spec.
 message DatabaseServiceSpecV1 {
   // ResourceMatchers is the configured match for Database resources.
-  repeated wrappers.LabelValues ResourceMatchers = 1 [
-    (gogoproto.jsontag) = "resources",
+  repeated ResourceMatcher ResourceMatchers = 1 [(gogoproto.jsontag) = "resources"];
+}
+
+// ResourceMatcher is a set of properties that is used to match on resources.
+message ResourceMatcher {
```
[the remark this answers is not in the document]
Comment (marcoandredinis) [reply api/proto/teleport/legacy/types/types.proto:4436]: There are other systems using this same representation of ResourceMatchers
For example, `service.KubeConfig.ResourceMatchers`
However, I do agree that we should keep them separate at the Proto level

**Review by codingllama — APPROVED** [review]

**Pull request merged**